app/src/main/java/com/kickstarter/viewmodels/MessageThreadsViewModel.java

```diff
@@ -58,9 +66,12 @@ public ViewModel(final @NonNull Environment environment) {
       this.client = environment.apiClient();
       this.currentUser = environment.currentUser();
 
-      final Observable<User> freshUser = this.client.fetchCurrentUser()
+      final Observable<User> freshUser = intent()
+        .compose(takeWhen(this.onResume))
```

Comment:
this is the first place in which we've used `onResume()` as a lifecycle input!! 

the reasoning was that I wanted a signal similar to iOS's `viewDidAppear` to refresh the current user appropriately...the downside is that we'll be refreshing the user on rotation even, but I figured that if we tie many of our api calls to lifecycle events such as `viewWillAppear` and `viewDidAppear` in iOS this may be fine?

other option is to store a `BooleanPreference` on the environment which i'm chill with but wanted to try this approach and see how y'all feel